Content.Server/Administration/Commands/FindEntitiesWithComponents.cs

```diff
@@ -45,10 +46,12 @@ public void Execute(IConsoleShell shell, string argStr, string[] args)
             }
 
             var entityManager = IoCManager.Resolve<IEntityManager>();
-            var query = new MultipleTypeEntityQuery(components);
             var entityIds = new HashSet<string>();
 
-            foreach (var entity in entityManager.GetEntities(query))
+            var entitiesWithComponents = components.Select(c => entityManager.ComponentManager.GetAllComponents(c).Select(x => x.Owner));
+            var entitiesWithAllComponents = entitiesWithComponents.Skip(1).Aggregate(new HashSet<IEntity>(entitiesWithComponents.First()), (h, e) => { h.IntersectWith(e); return h; });
```

Comment:
I guess it's okay for a debug command. I assume you know LINQ is generally to be avoided in games.

Reply:
Thanks for reminding me to watch out for stuff like that, since I forgot gaming is a bit different than business software.

Reply:
The occasional ToArray or ToList is okay (same with AsParallel()) but chaining a bunch allocates out the ass and often shows up on a profiler.